variants/devise/template.rb

```diff
@@ -0,0 +1,137 @@
+# Allow us to copy file with root at the directory this file is in
+source_paths.unshift(File.dirname(__FILE__))
+
+##
+# This template can be used either as part of starting a new Rails project or
+# by running:
+#
+#     bundle exec rails app:template LOCATION=http://path/to/this/template.rb
+#
+def ask_with_default(question, color, default)
+  question = (question.split("?") << " [#{default}]?").join
+  answer = ask(question, color)
+  answer.to_s.strip.empty? ? default : answer
+end
+
+def print_header(msg)
+  puts "=" * 80
+  puts msg
+  puts "=" * 80
+end
+
+print_header "Adding devise to Gemfile"
+run "bundle add devise"
+
+print_header "Running devise generator"
+run "bundle exec rails generate devise:install"
+
+print_header "Generating User model with devise"
+run "bundle exec rails generate devise User"
+
+print_header "Running db migration"
+run "bundle exec rails db:migrate"
+
+print_header "Copying devise views into the application"
+run "bundle exec rails generate devise:views users"
+
+##
+# Tweak the generated devise config file
+#
+print_header "Tweaking config/initializers/devise.rb"
+
+mail_sender =  ask_with_default("What 'From:' addresss should this app use for emails from Devise?", :blue, "change-me@example.com")
```

Comment:
:elephant: it might be better to do this at the `ActionMailer::Base.default` level, since we rarely, if ever want to send Devise mail from a different address than other transactional email.

Reply:
I normally do `ENV["MAIL_FROM"]` with a default